tests/run/i20145.scala

```diff
@@ -0,0 +1,15 @@
+import scala.annotation.tailrec
+@tailrec
+def foo(i: Int): Int = {
+  if (i > 10000000) {
+    i
+  } else {
+    val bar: String = {
+      return foo(i + 1)
+      "foo"
+    }
+    -1
+  }
+}
+@main def Test =
+  println(foo(0))
```

Comment:
Unrelatedly, you can `assert(foo(0) == K)` to avoid introducing a check file.

Also, a PR can be in "draft" mode to mean "not ready for review". If you accidentally push to your branch while a PR is closed, it doesn't let you re-open the PR (for some reason).